sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -20,247 +19,399 @@
+This function stores the data in the accumulator matrix from the 2d
+tiles back to memory.
 
-The base pointer `dest` here determines the starting address of the matrix to be stored. `Layout` determines whether the data is being written in a row (`row_major`), column major (`column_major`) fashion. `stride` describes the number of elements between consecutive rows for the row major layout, or between columns for the column major layout. 
+The base pointer `dest` here determines the starting address of the
+matrix to be stored. `Layout` determines whether the data is being
+written in a row (`row_major`), column major (`column_major`)
+fashion. `stride` describes the number of elements between consecutive
+rows for the row major layout, or between columns for the column major layout.
 
 
-#### Multiply and Add
+==== Multiply and Add
 
 ```c++
 namespace sycl::ext::oneapi::experimental::matrix {
-  template <typename Group, typename Ta, typename Tb, typename Tc, std::size_t M, std::size_t K, std::size_t N, 
+
+template <typename Group, typename Ta, typename Tb, typename Tc,
+  std::size_t M, std::size_t K, std::size_t N,
             layout LayoutA, layout LayoutB>
-  joint_matrix<Group, Td, use::accumulator, M, N, layout::dynamic> joint_matrix_mad(Group sg,
-    joint_matrix<Group, Ta, use::a, M, K, layoutA> A,
-    joint_matrix<Group, Tb, use::b, K, N, layoutB> B,
+  joint_matrix<Group, Td, use::accumulator, M, N, layout::dynamic>
+  joint_matrix_mad(Group g,
+    joint_matrix<Group, Ta, use::a, M, K, LayoutA> A,
+    joint_matrix<Group, Tb, use::b, K, N, LayoutB> B,
     joint_matrix<Group, Tc, use::accumulator, M, N, layout::dynamic> C);
-}
+
+} // namespace sycl::ext::oneapi::experimental::matrix
 ```
-The matrix multiply and add function performs the multiply operation on the matrices `A` and `B`, accumulate the result with `C` and return the result.
+The matrix multiply and add function performs the multiply operation
+on the matrices `A` and `B`, accumulates the result with `C` and returns
+the result.
 
 
-#### Matrix Initialization: `joint_matrix_fill`
-The current interface presented above assumes that all the matrices are directly loaded from memory. This new function called `joint_matrix_fill`  makes it possible to multiply a matrix which is not directly loaded from memory but rather initialized directly in the register. On Intel AMX, if the initialization constant is zero, this would map to the `_tile_zero` intrinsic: 
+==== Matrix Initialization: `joint_matrix_fill`
+Unlike `joint_matrix_load` that assumes that all the matrices are
+directly loaded from memory, `joint_matrix_fill`  makes it possible to
+multiply a matrix which is not directly loaded from memory but rather
+initialized directly in the register. On Intel AMX, if the
+initialization constant is zero, this would map to the `_tile_zero`
+intrinsic. Note that the value type `Tv` must be convertible to the
+matrix elements type `T`.
```

Comment:
> Since we decided to change `joint_matrix_load` and `joint_matrix_store` to have just one template type `T`, should we do the same here also and remove `Tv`?

In the case of fill, values can have other types that are convertible to T, like filling bfloat16 matrix with `5`. we use `static_cast` in the implementation so users do not need to cast.

Reply:
OK, that makes sense, and it is consistent with `std::fill`.